lib/SILGen/SILGenApply.cpp

```diff
@@ -5547,6 +5547,24 @@ RValue SILGenFunction::emitLiteral(LiteralExpr *literal, SGFContext C) {
     builtinLiteralArgs = RValue(*this, {integerManaged}, ty);
     builtinInit = integerLiteral->getBuiltinInitializer();
     init = integerLiteral->getInitializer();
+  } else if (auto floatLiteral = dyn_cast<FloatLiteralExpr>(literal)) {
+
+    ManagedValue floatManaged =
+        ManagedValue::forUnmanaged(B.createFloatLiteral(
+            floatLiteral,
+            // Find me some other way:
```

Comment:
You could try to factor out some common subexpressions here into named variables